inc/Engine/Preload/Controller/CheckExcludedTrait.php

```diff
@@ -14,4 +14,33 @@ protected function is_excluded( string $url ) {
 		$excluded = str_replace( '/', '\/', get_rocket_cache_reject_uri() );
 		return (bool) preg_match( "/$excluded/", $url );
 	}
+
+	/**
+	 * Check if the url is excluded by using a filter.
+	 *
+	 * @param string $url url to check.
+	 * @return bool
+	 */
+	protected function is_excluded_by_filter( string $url ): bool {
+		/**
+		 * Regex to exclude URI from preload.
+		 *
+		 * @param string[] regexes to check
+		 */
+		$regexes = (array) apply_filters( 'rocket_preload_exclude_urls', [] );
+
+		if ( empty( $regexes ) ) {
+			return false;
+		}
+
+		$regexes = array_filter( $regexes, 'is_string' );
```

Comment:
Here we are looping on the items and get only strings from inside and in the following line we are looping again, why not using the second loop and skip the iteration if it's not a string, that's good from performance wise, what do u think?